src/server.c

```diff
@@ -4076,6 +4078,15 @@ int prepareForShutdown(int flags) {
 
     /* Close the listening sockets. Apparently this allows faster restarts. */
     closeListeningSockets(1);
+
+    /* Close the module pipes to avoid fd leaks. */
+    closeModuleBlockedPipe();
+
+    /* Close all connections to master/replica nodes and sentinels. */
+    if (server.sentinel_mode) {
+        sentinelReleaseInstanceConnections(NULL, 1);
+    }
+
```

Comment:
This function (`prepareForShutdown`) is only called before `exit`, so there's no leak possible.
The comment says that the only reason we close the listening socket, so for faster restarts (not sure why).
other than that, redis counts on the OS to clean up after it, so i think we should remove that new block of code (it's comments are certainly false)

Reply:
In that case, the code of `sentinelReleaseInstanceConnections` function should also be erased?

Reply:
If closing sockets makes the restart faster, isn't it (sentinelReleaseInstanceConnections) supposed to achieve the same effect?

Reply:
i think that only applies for listening sockets.
the commit that added it isn't very specific 80e87a461a7d2ee54430127bdb6ac014778c54c5 but i suppose it means that then new process that comes up is able to bind successfully immediately.
i don't think it applies for other (non listen) sockets

Reply:
so, yeah, if that's the only use for the new `sentinelReleaseInstanceConnections` let's drop that function too.

Reply:
Done.